graph2tac/loader/data_server.py

```diff
@@ -171,23 +171,67 @@ def cluster_to_graph(self, cluster):
             ], dtype = self._def_name_dtype)
         )
 
-    
+class Splitter:
+    def datapoint(self, d : Definition):
+        raise Exception("Not implemented")
+    def cluster(self, d : list[Definition]):
+        raise Exception("Not implemented")
+
+class SplitByHash(Splitter):
+    def __init__(self, split = (8,1,1), random_seed = 0):
+        self.data_server = None # to be set by data_server
+        self.split = split
+        self.random_seed = random_seed
+    def datapoint(self, d : Definition):
+        # to make it identical to vasily's loader
+        ident_64 = np.array(d.node.identity, dtype = np.uint64).item()
+        return get_split_label(ident_64, self.split, self.random_seed)
+    def cluster(self, d : list[Definition]):
+        return 0
```

Comment:
Why are clusters always part of training?

Reply:
That is the old behavior

Reply:
Was there an actual historical reason for this, or is this just a bug? If the latter, I'd fix it...

Reply:
to me, it is a logical behavior

Reply:
when you randomly split on lemmata

Reply:
what other split on definitions do you imagine?

Reply:
Ah, okay. I misunderstood the purpose of `datapoint` and `cluster`. I get what is happening now, but I was interpreting them as doing exactly the same, except that one took a list. Perhaps rename them to `lemma` and `definition_cluster`?

Reply:
I wouldn't `split.lemma_split` feels having unnecessary repetition to me.

Reply:
That is a fine point.  We can leave it as `lemma` and `definition cluster`.